### ISSUE TITLE
Reproducing AUC ROC on VinDR

### ISSUE THREAD

**Issue by kaustubhrb** [prithuls/MV-SWIN-T#4]
Dear authors,
Thank you for making the code for this work public. 

I wanted to seek some advice on reproducing the AUC score reported for the VinDR dataset. I'm currently getting an AUC in the range of 0.57 on the test set. Could you let me know how many epochs does the model take to converge on the VinDR dataset?

Thank you!

**Comment by sushmitaSARKER** [comment]
Thank you for your question. I believe, we ran the model for 200 epochs.

**Comment by kaustubhrb** [comment]
Thank you! I'll try this out. 

Also, I have implemented my own data loader. Can you let me know for VinDR, did you end up using the breast BIRADS or the Findings BIRADS for computing your ground truth label? I'm taking breast BIRADS at the moment where BIRADS 1,2,3 are taken as benign and 4,5 are taken as malignant. Can you confirm if this aligns with your setup of the data loader?

**Comment by sushmitaSARKER** [comment]
Yes, we also used the breast BIRADS scores for computing the ground truth labels. Our setup was the same — BIRADS 1, 2, and 3 were considered benign, while BIRADS 4, 5 and 6 were considered malignant.

**Comment by kaustubhrb** [comment]
Great! Thanks for confirming

**Comment by kaustubhrb** [comment]
Hi,
I trained the model for 200 epochs. But still, the test AUC does not exceed 0.60. I have uploaded my dataloder script, it is possible that I'm doing anything incorrect here?

Thank you!

```
import os
import torch
from torch.utils.data import Dataset, DataLoader
from torchvision import utils, transforms
import pandas as pd
from PIL import Image 

class MakeDataset_VinDr_classification(Dataset):
    def __init__(self, image_dir, label_dir_csv, transform, mode, split_size, target_size):
        self.anns_csv = label_dir_csv
        self.image_dir = image_dir
        self.transforms = transform
        self.split=mode
        self.split_size = split_size
        self.target_size = target_size
        self.target_trasnforms = transforms

        self.df = pd.read_csv(self.anns_csv)
        print("Total entires in the CSV file: ", len(self.df))

        #Split the CSV as per the required split of the dataset
        chosen_ids = self.df[self.df['split'] == self.split]['record_id']
        if len(chosen_ids) == 0:
            print(f"Invalid split chosen: {mode}")
            exit()
        self.df = self.df[self.df['record_id'].isin(chosen_ids)]
        print(f"Total entires in {self.split}: ", len(self.df))

    def __len__(self):
        return len(self.df)
    
    def __getitem__(self, index):
        row = self.df.iloc[index]
        
        # Extract file paths for each image view
        cc_img_path = row['2d_cc']
        mlo_img_path = row['2d_mlo']
        
        cc_img = Image.open(cc_img_path).convert("RGB")
        mlo_img = Image.open(mlo_img_path).convert("RGB")        

        cc_tensor = self.transforms(cc_img)
        mlo_tensor = self.transforms(mlo_img)

        birads = row['asses'] #Breast BIRADS
        
        label = torch.tensor([1]) if birads >= 4 else torch.tensor([0])
        
        return cc_tensor, mlo_tensor, label
        

if __name__ == '__main__':
    ## Data Paths
    image_dir = "./Vindr-Mammo/vindr-mammo-a-large-scale-benchmark-dataset-for-computer-aided-detection-and-diagnosis-in-full-field-digital-mammography-1.0.0/images_jpg"
    label_dir_csv ="./processed_ann_files/vindr_processed.csv"
    target_size = 384
    ## Data Loaders

    transform = transforms.Compose([transforms.Resize((target_size, target_size)), transforms.ToTensor()])

    train_dataset= MakeDataset_VinDr_classification(image_dir = image_dir,
                                                            label_dir_csv = label_dir_csv,
                                                            transform=transform,
                                                            mode='test',
                                                            split_size= 0.2,
                                                            target_size= target_size)
    train_loader = DataLoader(train_dataset, batch_size=4, shuffle=True)

    for idx, (cc_img, mlo_img, label) in enumerate(train_loader):
        print(idx, cc_img.size(), mlo_img.size(), label.size())
        print(label)
        exit()
```

**Comment by sushmitaSARKER** [comment]
Thanks for sharing the details. We’re currently tied up with some other projects, but once we get time, we can run the script from our end and share the details and results with you.
